src/main/java/Duke.java

```diff
@@ -1,10 +1,251 @@
+
+    private static void firstGreet() {
+        System.out.println("Hello! I'm Duke!");
+        System.out.println("What can I do for you?");
+        System.out.println("-------------------------------------");
+    }
+
+    private static String createRemainingString(String[] inputWords) {
+        StringBuilder sb = new StringBuilder();
+        for (int i = 2; i < inputWords.length; i++) {
+            sb.append(inputWords[i]);
+            if (i < inputWords.length - 1) {
+                sb.append(" ");
+            }
+        }
+        return sb.toString();
+    }
+
+    private static void makeTaskDone(String[] inputWords, ArrayList<Task> tasks) {
+        for (int i = 1; i < inputWords.length; i++) {
+            try {
+                int num = Integer.parseInt(inputWords[i]) - 1;
+                tasks.get(num).markAsDone();
+                System.out.println("Congrats! You've completed this task: [X] "
+                        + tasks.get(num).getTaskDesc());
+            } catch (NumberFormatException | IndexOutOfBoundsException e) {
+                System.out.println(SOMETHING_WRONG +
+                        "Please ensure that you have entered the index of the task(s) to be marked as done");
+                break;
+            }
+        }
+    }
+
+    private static void listAllTasks(ArrayList<Task> tasks) {
+        for (int i = 0; i < tasks.size(); i++) {
+            StringBuilder sb = new StringBuilder();
+            String done = tasks.get(i).isPending() ? "[ ]" : "[X]";
+            sb.append(i + 1).append(": ").append(tasks.get(i).getType())
+                    .append(done).append(" ").append(tasks.get(i).getTaskDesc());
+            if (!tasks.get(i).getAdditionalInfo().isEmpty()) {
+                sb.append(" ").append(tasks.get(i).getAdditionalInfo());
+            }
+            System.out.println(sb);
+        }
+    }
+
+    private static int extractIndexToModify(String[] inputWords) {
+
+        int num = -1;
+        try {
+            num = Integer.parseInt(inputWords[1]) - 1;
+        } catch (ArrayIndexOutOfBoundsException | NumberFormatException e) {
+            System.out.println(SOMETHING_WRONG
+                    + "Please input the item number you wish to edit.");
+        }
+        return num;
+
+    }
+
+    private static void deleteTask(String[] inputWords, ArrayList<Task> tasks) {
+
+        for (int i = inputWords.length - 1; i > 0; i--) {
+            try {
+                int num = Integer.parseInt(inputWords[i]) - 1;
+                System.out.println("Task " + (num + 1) + ": "
+                        + tasks.get(num).getTaskDesc() + " has been deleted");
+                if (tasks.size() == 1) {
+                    tasks.clear();
+                } else {
+                    tasks.remove(num);
+                }
+                nextAdd--;
+            } catch (NumberFormatException | IndexOutOfBoundsException e) {
+                System.out.println(e);
```

Comment:
If you are already catching this exception why do you need to print it as well? perhaps only print the user friendly exception (next line) logging the technical one for you?